test/unit/cmap/wire_protocol/responses.test.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+import { BSON, MongoDBResponse, OnDemandDocument } from '../../../mongodb';
+
+describe('class MongoDBResponse', () => {
+  it('is a subclass of OnDemandDocument', () => {
+    expect(new MongoDBResponse(BSON.serialize({ ok: 1 }))).to.be.instanceOf(OnDemandDocument);
+  });
+
+  context('get isError', () => {
+    it('returns true when ok is 0', () => {
+      const doc = new MongoDBResponse(BSON.serialize({ ok: 0 }));
+      expect(doc.isError).to.be.true;
+    });
+
+    it('short circuits detection of $err, errmsg, code', () => {
```

Comment:
are we missing tests to check that `isError` returns true if `$err`, `errmsg` or `code` are present and truthy?

Reply:
Good call, previously the code was `document.ok === 0 || document.$err || document.errmsg || document.code` so it wasn't just that those properties existed but that they were truthy values. Do you think it is acceptable to change this to a has check?

Reply:
I can't think of a scenario where the server might return a response with one of these fields present with a falsey value.  I don't think it would muck up the implementation much to do something like the following, so I guess I don't see much of a reason not to leave the logic as-is:

```typescriopt
get isError(): boolean {
  return Boolean(
    this.ok ?? this.$err ?? this.errmsg ?? this.code
  )
}
```

Reply:
https://github.com/mongodb/mongo/blob/master/src/mongo/db/dbmessage.cpp#L159-L162

https://github.com/mongodb/mongo/blob/master/src/mongo/db/basic_types.idl#L442-L444

I have confidence in our change here to existence check, and at least currently we still run the error object through the same deserialization so we don't have to opt into binding these fields to a specific type.